Source/Dafny/Compilers/Compiler-python.cs

```diff
@@ -460,12 +460,53 @@ protected class ClassWriter : IClassWriter {
       if (!createBody) { return null; }
       if (isStatic || NeedsCustomReceiver(member)) { wr.WriteLine("@staticmethod"); }
       wr.Write($"def {name}(");
-      WriteFormals(formals, isStatic && !NeedsCustomReceiver(member), wr);
+      var sep = "";
+      WriteRuntimeTypeDescriptorsFormals(member, ForTypeDescriptors(typeArgs, member, lookasideBody), wr, ref sep, tp => tp.CompileName);
+      WriteFormals(formals, isStatic && !NeedsCustomReceiver(member), ref sep, wr);
       return wr.NewBlockPy("):", close: BlockStyle.Newline);
     }
 
     protected override string TypeDescriptor(Type type, ConcreteSyntaxTree wr, IToken tok) {
-      throw new UnsupportedFeatureException(Token.NoToken, Feature.RuntimeTypeDescriptors);
+      Contract.Requires(type != null);
+      Contract.Requires(tok != null);
+      Contract.Requires(wr != null);
+
+      var customName = false;
+      var name = type.NormalizeExpandKeepConstraints() switch {
+        // unresolved proxy; just treat as bool, since no particular type information is apparently needed for this type
+        BoolType or TypeProxy => "bool",
+        IntType or BitvectorType => "int",
+        RealType => "real",
+        UserDefinedType udt => udt.ResolvedClass switch {
+          TypeParameter tp => TypeParameterDescriptor(tp),
+          ClassDecl => "null",
```

Comment:
Wouldn't this be better as an exception too? Perhaps an `UnsupportedFeatureException`?

Reply:
No, that's how we handle it in the other targets as well.

Reply:
null actually expands to `_dafny.defaults.null`, which eventually returns `None`.

Reply:
I'm pretty sure, this is covered by the tests.